### PR TITLE
Fix/12077: Manage members - When removing a user offline from workspace, user is not crossed out

### PR DESCRIPTION
**Pull request by tienifr**
### Details

This PR fixes the issue that cause by this PR: https://github.com/Expensify/App/pull/12545

### Fixed Issues

$ https://github.com/Expensify/App/issues/12770
$ https://github.com/Expensify/App/issues/12770#issuecomment-1318412114

### Tests

1. Go to Settings
2. Go to Profile
3. Go to any workspace
4. Click Manage members
5. Turn off internet connection and remove the user
6. Select all member

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to Settings
2. Go to Profile
3. Go to any workspace
4. Click Manage members
5. Turn off internet connection and remove the user and verify that the user is crossed out and slightly greyed out
6. Select all member and verify that can't check Select all checkbox when there is only me on the list
7. Select a member and verify that can't select checkbox of the greyed out user 
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).

- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for expected offline behavior are in the `Offline steps` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] I verified that if a function's arguments changed that all usages have also been updated correctly
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/113963320/203536564-116e6267-4940-4c79-ad8f-80b07ef96a9b.mp4

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/113963320/203536688-484cd20f-f42e-4b0d-a9ac-115c8c068ed9.mp4

#### Mobile Web - Safari

https://user-images.githubusercontent.com/113963320/203548093-71b4b31c-8529-40f4-bd37-9408713e7d60.mp4

#### Desktop

https://user-images.githubusercontent.com/113963320/203539816-1b6725d0-08ec-42a9-bf98-a2aa4ac766df.mp4

#### iOS

https://user-images.githubusercontent.com/113963320/203544031-7b119ce0-067a-4443-b8db-0f1d1ae0acd9.mp4

#### Android

https://user-images.githubusercontent.com/113963320/203545933-d5451be0-ec9b-404e-b8bf-12f6bf36ab89.mp4

### CODE REVIEW

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 1]
```diff
@@ -301,7 +301,6 @@ class WorkspaceMembersPage extends React.Component {
         const removableMembers = [];
         let data = [];
         _.each(policyMemberList, (policyMember, email) => {
-            if (policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) { return; }
             if (email !== this.props.session.email && email !== this.props.policy.owner) {
                 removableMembers.push(email);
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:305]: Removing `(policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE)` will allow addition of those members to `removableMembers`. Will it not cause any problem?

How about adding `(policyMember.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE)` in the above if statement?

Comment (tienifr) [reply src/pages/workspace/WorkspaceMembersPage.js:305]: Hi @sobitneupane, I don't see any issue when removing removing `if (policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE)` statement.

Comment (sobitneupane) [reply src/pages/workspace/WorkspaceMembersPage.js:305]: Removing `(policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE)` will allow addition of pending deleted members to `removableMembers`. I think crossed members deleted offline can be selected if you remove this line.

Comment (tienifr) [reply src/pages/workspace/WorkspaceMembersPage.js:305]: @sobitneupane  Thanks for pointing out it. I am fixing and testing again locally

Comment (tienifr) [reply src/pages/workspace/WorkspaceMembersPage.js:305]: oh, I saw the problem, I'll update my PR. Thanks @sobitneupane

Comment (tienifr) [reply src/pages/workspace/WorkspaceMembersPage.js:305]: @sobitneupane I fixed your comment, please help to check the change code and steps to test.

**Review by sobitneupane — COMMENTED** [review]
I can still select already removed account. It can be selected from the checkbox of the optionrow but not with Select all checkbox.

https://user-images.githubusercontent.com/25876548/203330910-8f2c9d2a-6e0f-4871-a7a8-6e752ff5933b.mov

I don't think we want to allow user to select already removed account. cc: @Julesssss

**Comment by tienifr** [comment]
@sobitneupane Thanks for pointing out this issue. My solution is to create a new field `pendingDelete` and add conditional for checkbox: `isChecked={_.contains(this.state.selectedEmployees, item.login) && item.pendingDelete === false}`
I just pushed this solution in a new commit. Please help to check. Thanks

**Review by sobitneupane — COMMENTED** [review]
Though checkbox remains unchecked, Remove button becomes clickable 
> https://user-images.githubusercontent.com/25876548/203481287-f9450e15-d814-48be-b5bf-54b39e97a4da.mov

**Comment by tienifr** [comment]
@sobitneupane  I just added the condition in the `onPress` function, we will not trigger `toggleUser` function if the user is pendingDelete. Please tell me if you have any concerns about that

###Result


https://user-images.githubusercontent.com/113963320/203488080-4f4e318a-823d-4109-a883-814666e27dff.mp4

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 2]
```diff
@@ -256,22 +256,22 @@ class WorkspaceMembersPage extends React.Component {
...
+                        onPress={() => item.pendingDelete === false && this.toggleUser(item.login)}
                         activeOpacity={0.7}
                     >
                         <CheckboxWithTooltip
                             style={[styles.peopleRowCell]}
-                            isChecked={_.contains(this.state.selectedEmployees, item.login)}
-                            disabled={!canBeRemoved}
-                            onPress={() => this.toggleUser(item.login)}
+                            isChecked={_.contains(this.state.selectedEmployees, item.login) && item.pendingDelete === false}
+                            disabled={!canBeRemoved || item.pendingDelete === true}
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:265]:
```suggestion
                            disabled={!canBeRemoved}
```

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 3]
```diff
@@ -256,22 +256,22 @@ class WorkspaceMembersPage extends React.Component {
...
+                            onPress={() => item.pendingDelete === false && this.toggleUser(item.login)}
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:266]:
```suggestion
                            onPress={() => this.toggleUser(item.login, item.pendingAction)}
```

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 4]
```diff
@@ -256,22 +256,22 @@ class WorkspaceMembersPage extends React.Component {
...
                             toggleTooltip={this.state.showTooltipForLogin === item.login}
                             text={this.props.translate('workspace.people.error.cannotRemove')}
                         />
                         <View style={styles.flex1}>
                             <OptionRow
-                                onSelectRow={() => this.toggleUser(item.login)}
+                                onSelectRow={() => item.pendingDelete === false && this.toggleUser(item.login)}
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:272]:
```suggestion
                                onSelectRow={() => this.toggleUser(item.login, item.pendingAction)}
```

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 5]
```diff
@@ -256,22 +256,22 @@ class WorkspaceMembersPage extends React.Component {
...
                                 forceTextUnreadStyle
-                                isDisabled={!canBeRemoved}
+                                isDisabled={!canBeRemoved || item.pendingDelete === true}
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:274]:
```suggestion
                                isDisabled={!canBeRemoved}
```

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 6]
```diff
@@ -301,20 +301,20 @@ class WorkspaceMembersPage extends React.Component {
         const removableMembers = [];
         let data = [];
         _.each(policyMemberList, (policyMember, email) => {
-            if (policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) { return; }
-            if (email !== this.props.session.email && email !== this.props.policy.owner) {
+            if (email !== this.props.session.email && email !== this.props.policy.owner && policyMember.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
                 removableMembers.push(email);
             }
             const details = lodashGet(this.props.personalDetails, email, {displayName: email, login: email, avatar: Expensicons.FallbackAvatar});
+            const pendingDelete = policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:308]:
```suggestion
```

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 7]
```diff
@@ -301,20 +301,20 @@ class WorkspaceMembersPage extends React.Component {
...
             data.push({
                 ...policyMember,
                 ...details,
+                pendingDelete,
```
Comment (sobitneupane) [inline src/pages/workspace/WorkspaceMembersPage.js:312]:
```suggestion
```

**Review by sobitneupane — COMMENTED** [review]
Changes test well. To clean up the code, i have suggested few changes. If it works well, let's make the changes and complete the PR.
Suggested few changes.
```diff
-        const canBeRemoved = this.props.policy.owner !== item.login && this.props.session.email !== item.login;
+        const canBeRemoved = this.props.policy.owner !== item.login && this.props.session.email !== item.login && item.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
```

```diff
-    toggleUser(login) {
-        if (this.willTooltipShowForLogin(login)) {
+    toggleUser(login, pendingAction) {
+        if (this.willTooltipShowForLogin(login) || pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
            return;
        }
```

**Comment by tienifr** [comment]
Hi @sobitneupane I just fixed as your suggestion and completing screenshots for this PR

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/203614177-897f872e-8658-4ebf-b7ca-0f90f5bf0671.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/25876548/203614352-69125107-1039-430c-9dbd-c4fa2492580e.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/25876548/203614552-1d83e67e-84f8-4255-a901-3c3e9f795fa2.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/203614292-9b1e55cc-874f-411f-8463-fef0098f13f6.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/203614479-60288c45-8c3b-41f4-a026-fe1f9f59b41b.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/25876548/203614457-4d44518d-3e92-43e3-8efa-b6c385f166fe.mov

</details>

**Review by Julesssss — APPROVED** [review]

**Pull request merged**